Odroid h4 netcard support

miczyg1 opened the pull request:
UNTESTED

miczyg1 committed:
3rdparty/dasharo-blobs: Bump for ODROID netcard support

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
mb/hardkernel/odroid-h4: Add netcard support

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
configs/config.hardkernel_odroid_h4: Enable quiet and fast boot

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
configs/config.hardkernel_odroid_h4: Enable ME configuration menu

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
vendorcode/dasharo,soc/intel: Add IBECC option

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
mainboard/hardkernel/odroid-h4: Add IBECC configuration

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
payloads/external/edk2: Add IBECC option support

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
configs/{msi,hardkernel}: Update configs for new memory menu PCDs

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

frafra commented on src/mainboard/hardkernel/odroid-h4/Kconfig:
```diff
@@ -80,4 +80,19 @@ config ODROID_H4_ENABLE_SAGV
+	default n
+	help
+	  Enables 4x1 Pcie netcard support in the project. Enable this
+	  option when hte M.2 NVMe slot is bifurcated as 4x1 to support
```
Typo? `hte` -> `the`

miczyg1 replied:
Yes, thanks for pointing out.

miczyg1 replied:
Fixed: https://github.com/Dasharo/coreboot/compare/1974615623289c35bc71e39a72f42f3db595928f..85f42bde76c468fa83da34577996a82b6158682f

miczyg1 committed:
mb/hardkernel/odroid-h4/data.vbt: Update VBT for newest FSP GOP

Fixes the issue with display not working in the firmware.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
The build works on Ubuntu 24.04 properly. Can get IP address on all 4 ports and ping a google.com domain.

miczyg1 committed:
mb/hardkernel/odroid-h4/devicetree.cb: Disable DSP

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
Windows also does not show any problems. All NICs are working well and can browse sites when Ethernet is plugged into each port.

miczyg1 commented:
[odroid-netcard_2025_06_11_11_25_42.zip](https://github.com/user-attachments/files/20688369/odroid-netcard_2025_06_11_11_25_42.zip)

Logs from tests. I made 2 loopbacks and tested with iperf3 two pairs on the netcard. The speed was close to 2.5G so it is safe to assume that it really works.

miczyg1 committed:
configs/config.hardkernel_odroid_h4: Enable FW performance measurement

FW performance measurement is required to measure difference in boot
time with fast boot.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
